### ISSUE TITLE
[12.][l10n_it_edi] Wrong Untaxed, Tax, Total calculation?

### ISSUE THREAD

**Issue by GSLabIt** [odoo/odoo#41511]
Impacted versions: 12.0 

Importing Electronic invoice we get wrong `Untaxed Amount`, `Tax` and `Total` values.

Steps to reproduce:
1. Upload attached file from Invoicing -> Bills -> UPLOAD

Current behavior:
If set `Round per line`, you get:
<img width="1206" alt="Schermata 2019-12-08 alle 11 30 07" src="https://user-images.githubusercontent.com/11652845/70388140-19701080-19ae-11ea-8eba-8f09c5694511.png">

If you set `Round Globally`:
<img width="1198" alt="Schermata 2019-12-08 alle 11 32 04" src="https://user-images.githubusercontent.com/11652845/70388153-5f2cd900-19ae-11ea-9e17-12c94c109172.png">

Expected behavior:
It's expected to have same value as in Electronic Invoice (validated by authority):
```
      <DatiGeneraliDocumento>
        <TipoDocumento>TD01</TipoDocumento>
        <Divisa>EUR</Divisa>
        <Data>2019-11-30</Data>
        <Numero>12009/A</Numero>
        <ImportoTotaleDocumento>622.60</ImportoTotaleDocumento>
        <Causale>Vendita articoli</Causale>
      </DatiGeneraliDocumento>

```
```
      <DatiRiepilogo>
        <AliquotaIVA>22.00</AliquotaIVA>
        <ImponibileImporto>510.33</ImponibileImporto>
        <Imposta>112.27</Imposta>
        <EsigibilitaIVA>I</EsigibilitaIVA>
      </DatiRiepilogo>

```
File to test:
[IT09252150017_0FWKC.xml.zip](https://github.com/odoo/odoo/files/3936107/IT09252150017_0FWKC.xml.zip)

Any hint?

Video/Screenshot link (optional):

**Comment by GSLabIt** [comment]
@pedrobaeza i think issue could be related to tax calculation in account, rather than localization itself

**Comment by jco-odoo** [comment]
@GSLabIt The system uses the price unit and the quantity to calculate the total amounts, and if you use fuel with a lot of decimals, rounding errors will occur.  Did you try adjusting the decimal precision on quantity and price? 

Of course, it is easy to say that the user can adapt the correct numbers afterwards as normally you will still check the invoice imported.  Maybe there is a way to import in a more stable way, but I don't see it immediately honestly.

**Comment by GSLabIt** [comment]
> @GSLabIt The system uses the price unit and the quantity to calculate the total amounts, and if you use fuel with a lot of decimals, rounding errors will occur. Did you try adjusting the decimal precision on quantity and price?
> 
> Of course, it is easy to say that the user can adapt the correct numbers afterwards as normally you will still check the invoice imported. Maybe there is a way to import in a more stable way, but I don't see it immediately honestly.

Yes, and it doesn't work properly, sometimes it works sometimes not. You also have issues with less decimal but discounts as discount is monetary instead of float. There are a lot of issues like this one need to be addressed.

So at the end of the day, you see that collaboration will be quite useful on both ways; making things auto install doesn't help to get it sorted out.

**Comment by parchanco** [comment]
Any solution or workaround for this? we are having the same issue.
